pkg/cmd/init.go

```diff
@@ -346,6 +345,33 @@ func initKubeconfigs(
 	if err != nil {
 		return err
 	}
+
+	u, err := url.Parse(cfg.Cluster.URL)
+	if err != nil {
+		return fmt.Errorf("failed to parse cluster URL: %v", err)
+	}
+	apiServerPort, err := cfg.Cluster.ApiServerPort()
+	if err != nil {
+		return fmt.Errorf("failed to get apiserver port: %v", err)
+	}
+
+	// If using localhost this is part of the default URL and it needs external access.
+	// Generate all other kubeconfigs, one per name.
+	if u.Hostname() == "localhost" || u.Hostname() == "127.0.0.1" {
```

Comment:
Hmm, I propose that we generate the alt kubeconfigs always, regardless of the URL/host we configure.
Unless I am missing something and this is also part of the SNI/TLS issues we have and it cannot be done.

Reply:
You are right. I was assuming that a non-default configuration should dictate a single kubeconfig and disregard the rest of the alternative names in the `server` field, but there might be use cases where you need to refer to the same cluster with different names coming from different DNS servers.

Reply:
Done